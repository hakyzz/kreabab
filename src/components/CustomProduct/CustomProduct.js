import React, { Component } from 'react';
import './CustomProduct.css';
import { categories } from '../../defs/kreabab';
import FleischTopping from '../../assets/img/toppings/fleisch.png';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

class CustomProduct extends Component {

    render() {

        const bread = categories[0].items.find((item) => item.title === this.props.product.bread) || {};
        const toppings = categories[1].items.filter(topping => this.props.product.toppings.includes(topping.title));
        const sauces = categories[2].items.filter(sauce => this.props.product.sauces.includes(sauce.title));

        return (
            <div className="custom-product">

                <ReactCSSTransitionGroup
                    transitionName={{
                        enter: 'kreabab__topping--enter',
                        leave: 'kreabab__topping--leave',
                    }}
                    transitionEnterTimeout={300}
                    transitionLeaveTimeout={200}>
                    {[...toppings, ...sauces].map(item =>
                        <div key={item.title} className={`product-topping ${item.name === 'sauceType' ? 'product-topping--sauce' : ''}  `}>
                            <div className={`kreabab__topping kreabab_topping--${item.title.toLowerCase()}`} style={{backgroundImage: `url(${item.topping})`}}/>
                        </div>
                    )}
                </ReactCSSTransitionGroup>

                <div key="fleisch" className="product-topping">
                    <div className={`kreabab__topping kreabab__topping--fleisch`} style={{backgroundImage: `url(${FleischTopping})`}}/>
                </div>

                <img src={bread.topping} className="custom-product__image" alt={bread.title}/>
                <h2>{bread.title}</h2>
            </div>
        );
    }
}

export default CustomProduct;