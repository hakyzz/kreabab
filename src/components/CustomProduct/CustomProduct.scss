.custom-product {
  padding-top: 20px;
  text-align: center;

  &__image {
    width: 200px;
    padding-bottom: 20px;
  }

  .kreabab__topping {
    position: absolute;
    background-repeat: no-repeat;
    background-size: contain;
    transition-timing-function: cubic-bezier(1, 0, 0, 1);

    &--enter {
      transition: transform 0.3s, opacity 0.3s;
      transform: translateY(-200px);
      opacity: 0;
      &-active {
        transform: translateY(0);
        opacity: 1;
      }
    }

    &--leave {
      transition: transform 0.2s, opacity 0.2s;
      transform: translateY(0);
      opacity: 1;
      &-active {
        transform: translateY(-200px);
        opacity: 0;
      }
    }

    &--fleisch {
      transform: translateY(0);
      opacity: 1;
    }
  }

  .product-topping {
    position: relative;

    &--sauce {
      z-index: 99;
    }
  }
}

@media (min-width: 768px) {
  .custom-product {
    padding-top: 156px;

    &__image {
      width: 350px;
    }

    .kreabab__topping {
      position: absolute;
      left: 50%;
      transform: translateX(-50%);
      width: 350px;
      padding-top: 20px;
      height: 350px;
      background-size: contain;
      background-repeat: no-repeat;
    }
  }
}