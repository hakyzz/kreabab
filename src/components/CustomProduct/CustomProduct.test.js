import React from 'react';
import ReactDOM from 'react-dom';
import CustomProduct from './CustomProduct';
import { shallow } from 'enzyme';

const props = {
    product: {
        bread: "Fladenbrot",
        sauces: [],
        toppings: []
    }
};

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CustomProduct {...props} />, div);
});

it('it matches snapshop', () => {
    const _props = { ...props };
    const tree = shallow(<CustomProduct {...props} />);
    expect(tree).toMatchSnapshot();
});