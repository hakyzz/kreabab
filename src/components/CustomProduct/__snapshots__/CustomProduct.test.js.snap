// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomProduct
    product={
        Object {
            "bread": "Fladenbrot",
            "sauces": Array [],
            "toppings": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CSSTransitionGroup
          transitionAppear={false}
          transitionEnter={true}
          transitionEnterTimeout={300}
          transitionLeave={true}
          transitionLeaveTimeout={200}
          transitionName={
                    Object {
                              "enter": "kreabab__topping--enter",
                              "leave": "kreabab__topping--leave",
                            }
          }
>
          
</CSSTransitionGroup>,
        <div
          className="product-topping"
>
          <div
                    className="kreabab__topping kreabab__topping--fleisch"
                    style={
                              Object {
                                        "backgroundImage": "url(fleisch.png)",
                                      }
                    }
          />
</div>,
        <img
          alt="Fladenbrot"
          className="custom-product__image"
          src="fladenbrot.png"
/>,
        <h2>
          Fladenbrot
</h2>,
      ],
      "className": "custom-product",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "transitionAppear": false,
          "transitionEnter": true,
          "transitionEnterTimeout": 300,
          "transitionLeave": true,
          "transitionLeaveTimeout": 200,
          "transitionName": Object {
            "enter": "kreabab__topping--enter",
            "leave": "kreabab__topping--leave",
          },
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "fleisch",
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="kreabab__topping kreabab__topping--fleisch"
            style={
                        Object {
                                    "backgroundImage": "url(fleisch.png)",
                                  }
            }
/>,
          "className": "product-topping",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "kreabab__topping kreabab__topping--fleisch",
            "style": Object {
              "backgroundImage": "url(fleisch.png)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Fladenbrot",
          "className": "custom-product__image",
          "src": "fladenbrot.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fladenbrot",
        },
        "ref": null,
        "rendered": "Fladenbrot",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CSSTransitionGroup
            transitionAppear={false}
            transitionEnter={true}
            transitionEnterTimeout={300}
            transitionLeave={true}
            transitionLeaveTimeout={200}
            transitionName={
                        Object {
                                    "enter": "kreabab__topping--enter",
                                    "leave": "kreabab__topping--leave",
                                  }
            }
>
            
</CSSTransitionGroup>,
          <div
            className="product-topping"
>
            <div
                        className="kreabab__topping kreabab__topping--fleisch"
                        style={
                                    Object {
                                                "backgroundImage": "url(fleisch.png)",
                                              }
                        }
            />
</div>,
          <img
            alt="Fladenbrot"
            className="custom-product__image"
            src="fladenbrot.png"
/>,
          <h2>
            Fladenbrot
</h2>,
        ],
        "className": "custom-product",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "transitionAppear": false,
            "transitionEnter": true,
            "transitionEnterTimeout": 300,
            "transitionLeave": true,
            "transitionLeaveTimeout": 200,
            "transitionName": Object {
              "enter": "kreabab__topping--enter",
              "leave": "kreabab__topping--leave",
            },
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "fleisch",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="kreabab__topping kreabab__topping--fleisch"
              style={
                            Object {
                                          "backgroundImage": "url(fleisch.png)",
                                        }
              }
/>,
            "className": "product-topping",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "kreabab__topping kreabab__topping--fleisch",
              "style": Object {
                "backgroundImage": "url(fleisch.png)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Fladenbrot",
            "className": "custom-product__image",
            "src": "fladenbrot.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fladenbrot",
          },
          "ref": null,
          "rendered": "Fladenbrot",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
